### ISSUE TITLE
Licence check

### ISSUE THREAD

**Issue by mdevilliers** [thingful/device-hub#99]
This is probably something I don't understand -

```
1.47s$ make check-vendor-licenses
[      OK]  Found accepted license        license = MIT             direct dependency = yes  package = github.com/asaskevich/govalidator
[      OK]  Found accepted license        license = MIT             direct dependency = yes  package = github.com/mdevilliers/go-compose
[      OK]  Found accepted license        license = MIT             direct dependency = yes  package = github.com/natefinch/lumberjack
[      OK]  Found accepted license        license = MIT             direct dependency = yes  package = github.com/rs/xid
[      OK]  Found accepted license        license = MIT             direct dependency = yes  package = github.com/spf13/cast
[      OK]  Found accepted license        license = Apache-2.0      direct dependency = no   package = cloud.google.com/go
[      OK]  Found accepted license        license = MIT             direct dependency = no   package = github.com/boltdb/bolt
[ WARNING]  Could not guess license type  license = unknown         direct dependency = no   package = github.com/davecgh/go-spew
[ WARNING]  Criteria for license unknown  license = EPL-1.0         direct dependency = no   package = github.com/eclipse/paho.mqtt.golang
[      OK]  Found accepted license        license = NewBSD          direct dependency = no   package = github.com/fiorix/protoc-gen-cobra
[      OK]  Found accepted license        license = NewBSD          direct dependency = no   package = github.com/golang/protobuf
[      OK]  Found accepted license        license = Apache-2.0      direct dependency = no   package = github.com/inconshreveable/mousetrap
[      OK]  Found accepted license        license = MIT             direct dependency = no   package = github.com/kelseyhightower/envconfig
[ WARNING]  Criteria for license unknown  license = FreeBSD         direct dependency = no   package = github.com/pmezard/go-difflib
[      OK]  Found accepted license        license = MIT             direct dependency = no   package = github.com/robertkrimen/otto
[      OK]  Found accepted license        license = MIT             direct dependency = no   package = github.com/sirupsen/logrus
[      OK]  Found accepted license        license = MIT             direct dependency = no   package = github.com/speps/go-hashids
[      OK]  Found accepted license        license = Apache-2.0      direct dependency = no   package = github.com/spf13/cobra
[      OK]  Found accepted license        license = NewBSD          direct dependency = no   package = github.com/spf13/pflag
[      OK]  Found accepted license        license = MIT             direct dependency = no   package = github.com/stretchr/testify
[      OK]  Found accepted license        license = NewBSD          direct dependency = no   package = golang.org/x/net
[      OK]  Found accepted license        license = NewBSD          direct dependency = no   package = golang.org/x/oauth2
[      OK]  Found accepted license        license = NewBSD          direct dependency = no   package = golang.org/x/sys
[      OK]  Found accepted license        license = NewBSD          direct dependency = no   package = golang.org/x/text
[      OK]  Found accepted license        license = Apache-2.0      direct dependency = no   package = google.golang.org/appengine
[      OK]  Found accepted license        license = Apache-2.0      direct dependency = no   package = google.golang.org/genproto
[      OK]  Found accepted license        license = NewBSD          direct dependency = no   package = google.golang.org/grpc
[ WARNING]  Criteria for license unknown  license = FreeBSD         direct dependency = no   package = gopkg.in/sourcemap.v1
[      OK]  Found accepted license        license = Apache-2.0      direct dependency = no   package = gopkg.in/yaml.v2
```
In .licences.yaml

```
accepted:
- MIT
- Apache-2.0
- NewBSD

unaccepted:
- GPL-2.0
- GPL-3.0

exceptions:
- github.com/davecgh/go-spew # Uses ISC license (https://opensource.org/licenses/ISC)
```

Why do the projects using EPL-1.0  and FreeBSD not fail the licence check?

**Comment by chespinoza** [comment]
Because there aren't specific criteria defined for these licenses, that's why the output is a warning.
If you want to reject FreeBSD licenses then it should be added under the unaccepted tag,
in the same way, if you don't want a warning then the license should be added to accepted licenses.

**Comment by mdevilliers** [comment]
I understand now.

Could the licence checker have a "fail on warning" mode? 

My use case is that I don't want to rely on having to look at the output from the build - I would rather it tells me that we have an unaccounted for license.

**Comment by chespinoza** [comment]
I've just added a `-strict` mode for goliscan, if we do use that flag in the makefile all warnings will make fail the check.
The problem I see is that there is a dependency that has an unknown license and it will fail.
In the other hand, in order to pass _FreeBSD_ and _EPL-1.0_ cases both should be added under the **accepted** classification in the .licenses.yaml file

**Comment by mdevilliers** [comment]
Awesome - could you make the changes (add strict mode, FreeBSD and EPL are OK) on a branch and we can try and resolve the last issue? 

I did have a quick check and couldn't see any branch with the changes but I might have missed it.

Thanks

**Comment by chespinoza** [comment]
It's done in _check-vendor-licenses_ branch, and the build failing on that point.
Now let's figure out what's happens there.

**Comment by chespinoza** [comment]
Addressed in #100
